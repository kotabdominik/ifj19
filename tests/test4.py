a = 0
b = 441
print('budes stuck v cyklu, dokud neuhadnes cislo v rozmezi 0 - 1 000')
while a != b:
  a = inputi()
  if a != b:
    print('zadana hodnota', a, 'neni pozadovana hodnota')
  else:
    print('nyni si pravy programator')