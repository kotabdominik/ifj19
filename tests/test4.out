.IFJcode19
JUMP $$MAIN
LABEL $ORD
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$CHR2INT
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
ADD LF@$TMPLEN LF@$TMPLEN int@-1
DEFVAR LF@$BOOLCHECK
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
STRI2INT LF@$RETVAL LF@$STRING LF@$POSIT
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORORD
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $BETA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL int
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $SUBSTR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$DELKA
POPS LF@$DELKA
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
DEFVAR LF@$BOOLCHECK
ADD LF@$TMPLEN LF@$TMPLEN int@-1
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
GT LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
EQ LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
GETCHAR LF@$RETVAL LF@$STRING LF@$POSIT
ADD LF@$POSIT LF@$POSIT int@1
DEFVAR LF@$TMPSTR
ADD LF@$DELKA LF@$DELKA LF@$POSIT
SUB LF@$DELKA LF@$DELKA int@1
LABEL $WHILE_LOOP_B
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$DELKA
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$TMPLEN
GETCHAR LF@$TMPSTR LF@$STRING LF@$POSIT
CONCAT LF@$RETVAL LF@$RETVAL LF@$TMPSTR
ADD LF@$POSIT LF@$POSIT int@1
JUMP $WHILE_LOOP_B
LABEL $WHILE_LOOP_E
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORSUB
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ALFA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL string
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $GAMA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL float
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $LENGTH
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$1
POPS LF@$1
STRLEN LF@$RETVAL LF@$1
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $CHAR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$ASC
POPS LF@$ASC
INT2CHAR LF@$RETVAL LF@$ASC
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ZEROERROR
EXIT int@9
LABEL $$MAIN
CREATEFRAME
PUSHFRAME
DEFVAR GF@$VARa
PUSHS int@0
POPS GF@$VARa
DEFVAR GF@$VARb
PUSHS int@441
POPS GF@$VARb
PUSHS string@budes\032stuck\032v\032cyklu,\032dokud\032neuhadnes\032cislo\032v\032rozmezi\0320\032-\0321\032000
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7fffe30e2fd4
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7fffe30e2fd4  string@None
POPFRAME
CREATEFRAME
PUSHFRAME
DEFVAR LF@$COND0x7fffe30e31f0
PUSHS GF@$VARa
PUSHS GF@$VARb
EQS
NOTS
LABEL WHILE$BEGIN$0x7fffe30e31f0
POPS LF@$COND0x7fffe30e31f0
JUMPIFNEQ WHILE$END$0x7fffe30e31f0 bool@true LF@$COND0x7fffe30e31f0
CALL $BETA
POPS GF@$VARa
CREATEFRAME
PUSHFRAME
PUSHS GF@$VARa
PUSHS GF@$VARb
EQS
NOTS
DEFVAR LF@$COND0x7fffe30e37a0
POPS LF@$COND0x7fffe30e37a0
JUMPIFNEQ ELSE0x7fffe30e37a0 bool@true LF@$COND0x7fffe30e37a0
PUSHS string@zadana\032hodnota
PUSHS GF@$VARa
PUSHS string@neni\032pozadovana\032hodnota
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7fffe30e3994
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
DEFVAR LF@$PEPEGA1
POPS LF@$PEPEGA1
DEFVAR LF@$PEPEGA2
POPS LF@$PEPEGA2
WRITE LF@$PEPEGA2
WRITE string@\032
WRITE LF@$PEPEGA1
WRITE string@\032
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7fffe30e3994  string@None
POPFRAME
JUMP END0x7fffe30e37a0
LABEL ELSE0x7fffe30e37a0
PUSHS string@nyni\032si\032pravy\032programator
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7fffe30e4244
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7fffe30e4244  string@None
POPFRAME
LABEL END0x7fffe30e37a0
POPFRAME
PUSHS GF@$VARa
PUSHS GF@$VARb
EQS
NOTS
JUMP WHILE$BEGIN$0x7fffe30e31f0
LABEL WHILE$END$0x7fffe30e31f0
POPFRAME
